hummingbot/hummingbot
# fix/ `hummingbot_quickstart.py` improvements when using Gateway connector

**Before submitting this PR, please make sure**:


Link to Snapshot Proposal: https://snapshot.org/#/hbot-prp.eth/proposal/0xa20c501250a110d4630bb96d04aa24c8ebc7c3d5a428f8e3927e18515f3b9049

**A description of the changes proposed in the pull request**:
1.  `hummingbot_quickstart.py` will now not require a confirmation prompt when starting a gateway-required strategy
2. `StartCommand` now waits for the gateway service to be `ONLINE` before initialising a gateway-required strategy.

**Tests performed by the developer**:
1. Add additional initialisation parameter for `HummingbotApplication` that will be used to determine if confirmation prompt is necessary
2. Add helper functions within the `StartCommand` class to do additional preliminary checks to ensure that Gateway service is `ONLINE`.
4. Refactor and add additional properties into `GatewayStatusMonitor` to help with the above checks. 
5. Re-define `GatewayStatus` Enum. `ONLINE` is now used to indicate that the Hummingbot client is able to receive a response from the Gateway Service, while `OFFLINE` is used to indicate otherwise.

**Tips for QA testing**:
- Running `hummingbot_quickstart.py` with a gateway-supported strategy would no longer be blocked by the confirmation prompt.
- Any subsequent manual inputs to `start` the bot, will still require confirmation from the user.
- Ensure that the bot now waits for the the gateway service to be running and online before the gateway-reliant strategy gets started. 
- A 3 minute timeout is enforced to prevent the bot from waiting indefinitely
